qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -60,23 +60,19 @@ def __init__(self, q_job, api, is_device):
         self._backend_name = self._qobj.get('config').get('backend_name')
         self._status = JobStatus.INITIALIZING
         self._future_submit = self._executor.submit(self._submit)
-        self._status_msg = None
+        self._status_msg = 'Job is initializing. Please, wait a moment.'
+        self._queue_position = None
         self._cancelled = False
         self._exception = None
         self._is_device = is_device
 
-    def result(self, timeout=None, wait=5):
-        # pylint: disable=arguments-differ
-        while self._status == JobStatus.INITIALIZING:
-            time.sleep(0.1)
-        this_result = self._wait_for_job(timeout=timeout, wait=wait)
+    # pylint: disable=arguments-differ
+    def result(self, timeout=60, wait=5):
+        self._wait_for_initialization(timeout=timeout)
+        result = self._wait_for_job(timeout=timeout, wait=wait)
```

Comment:
I think the expectation for `timeout` will be the maximum time for the `result` call to return. Perhaps we can instead start a clock here (`timeout_time = time.time() + timeout`) and pass `timeout_time` to `_wait_for_initialization` and `_wait_for_job`.